### PR TITLE
Set `User` and `Breadcrumb` from Map

### LINKED ISSUE #1330: Use `User` and `Breadcrumb` data classes from a Map
### Description

Context https://github.com/getsentry/team-mobile/issues/59

### PR DESCRIPTION
**Pull request by denrase**
#skip-changelog

## :scroll: Description

Use new Cocoa/Java API to set `User` and `Breadcrumb` from map.

## :bulb: Motivation and Context

Closes #1330

## :green_heart: How did you test it?

Set user and breadcrumb in sample app and run on iOS & Android.

https://sentry-sdks.sentry.io/issues/3903865215/?project=5428562&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=1h&stream_index=0

Bildschirmfoto 2023-05-16 um 11 41 04

Bildschirmfoto 2023-05-16 um 11 40 59

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 1]
```diff
@@ -287,86 +287,22 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
   }
 
   private fun setUser(user: Map<String, Any?>?, result: Result) {
-    if (user == null) {
+    if (user != null) {
+      val options = HubAdapter.getInstance().options as SentryAndroidOptions
```
Comment (marandaneto) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]:
```suggestion
      val options = HubAdapter.getInstance().options
```

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 2]
```diff
@@ -287,86 +287,22 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
...
-
     result.success("")
   }
 
   private fun addBreadcrumb(breadcrumb: Map<String, Any?>?, result: Result) {
-    if (breadcrumb == null) {
-      result.success("")
-      return
+    if (breadcrumb != null) {
+      val options = HubAdapter.getInstance().options as SentryAndroidOptions
```
Comment (marandaneto) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]:
```suggestion
      val options = HubAdapter.getInstance().options
```

**Comment by marandaneto** [comment]
@denrase there's a lint issue:

> Superfluous Disable Command Violation: SwiftLint rule 'cyclomatic_complexity' did not trigger a violation in the disabled region; remove the disable command (superfluous_disable_command)

Is there a lint fix for swift lint? can we add it to https://github.com/getsentry/sentry-dart/blob/c3b512677c8657d821db7d03a2ebf5bb9de7960b/.github/workflows/analyze.yml#L43-L45

**Comment by denrase** [comment]
@marandaneto It seams that the commit script is not run anymore. Seems that $GITHUB_HEAD_REF is null somehow even on a branch? Maybe because it's a draft PR? Didn't find this documented. Trying to run with different vars.

Bildschirmfoto 2023-05-16 um 14 51 32

File: .github/workflows/analyze.yml [hunk 3]
```diff
@@ -44,11 +44,20 @@ jobs:
...
+        if: ${{ inputs.package == 'flutter' }}
+        with:
+          args: --strict --fix
+
         # actions/checkout fetches only a single commit in a detached HEAD state. Therefore
         # we need to pass the current branch, otherwise we can't commit the changes.
         # GITHUB_HEAD_REF is the name of the head branch. GitHub Actions only sets this for PRs.
-      - run: ../scripts/commit-formatted-code.sh $GITHUB_HEAD_REF
-        if: env.GITHUB_HEAD_REF != null
+      - name: commit formatted code
```
Comment (marandaneto) [inline .github/workflows/analyze.yml]: If this is not working, lets revert the format step and fix the lining manually, raise a new issue for this.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**